src/ralph/backends/http/async_lrs.py

```diff
@@ -141,12 +161,8 @@ async def read(  # pylint: disable=too-many-arguments
         if not target:
             target = self.statements_endpoint
 
-        query_params = {}
-        if query.query:
-            query_params.update(query.query)
-
-        # Set chunk_size to limit parameter if not exist in the query
-        query_params.setdefault("limit", chunk_size)
```

Comment:
We should still set the `limit` query parameter to the `chunk_size`!

We could also log a warning if the `limit` has already been set, as the user probably expect the `limit` parameter to limit the maximum number of statements when it limits the number of statements returned by each request.

Reply:
By default, the limit is set to 0 according to the spec. In this case, the limit parameter will always exist in the query and the case with having chunk_size ase default does not longer make sense.

Reply:
To the LRS specification yes! But in this client here, we are diverting the behavior of the `limit` query parameter to let the user choose the `chunk_size`.
If we don't set the `limit` to the `chunk_size`, we are not chunk sizing anymore, right?